File: Dart_Source_Files/payload/Question.dart
// ignore_for_file: file_names

import 'Answer.dart';

class Question {
  final int questId;
  String question;
  String source;
  List<Answer> answers;

  Question(
      {required this.questId,
      required this.question,
      required this.source,
      required this.answers});

  factory Question.fromJson(Map<String, dynamic> json) {
    return Question(
        questId: json['idQuest'],
        question: json['question'],
        source: json['source'],
        answers: List<Answer>.from(
            (json['answers'] as Iterable).map((e) => Answer.fromJson(e))));
  }
}

File: Dart_Source_Files/payload/Question.js
import { Answer } from "./Answer";

class Question {
  
    /**
     * 
     * @param {number} questId 
     * @param {string} question 
     * @param {string} source 
     * @param {Answer[]} answers 
     */
    constructor(questId, question, source, answers) {

        this.questId = questId;
        this.question = question;
        this.source = source;
        this.answers = answers;
    }

    /**
     * 
     * @param {object} json 
     * @returns {Question}
     */
    static fromJson(json) {

        const parsedAnswers = json.answers.map((item) => Answer.fromJson(item));

        return new Question(json.questId, json.question, json.source, parsedAnswers);
    }
}

export {Question};